ui/src/app/workflows/components/submit-workflow-panel.tsx

```diff
@@ -17,119 +17,92 @@ interface Props {
     workflowParameters: Parameter[];
 }
 
-interface State {
-    entrypoint: string;
-    entrypoints: string[];
-    parameters: Parameter[];
-    workflowParameters: Parameter[];
-    selectedTemplate: Template;
-    templates: Template[];
-    labels: string[];
-    error?: Error;
-    isSubmitting: boolean;
-}
-
 const workflowEntrypoint = '<default>';
 
-export class SubmitWorkflowPanel extends React.Component<Props, State> {
-    constructor(props: any) {
-        super(props);
-        const defaultTemplate: Template = {
-            name: workflowEntrypoint,
-            inputs: {
-                parameters: []
-            }
-        };
-        const state: State = {
-            entrypoint: workflowEntrypoint,
-            entrypoints: this.props.templates.map(t => t.name),
-            selectedTemplate: defaultTemplate,
-            parameters: [] as Parameter[],
-            workflowParameters: JSON.parse(JSON.stringify(this.props.workflowParameters)),
-            templates: [defaultTemplate].concat(this.props.templates),
-            labels: ['submit-from-ui=true'],
-            isSubmitting: false
-        };
-        this.state = state;
-    }
+export function SubmitWorkflowPanel(props: Props) {
+    const defaultTemplate: Template = {
+        name: workflowEntrypoint,
+        inputs: {
+            parameters: []
+        }
+    };
```

Comment:
this didn't change, but this looks like it's static. we can hoist this to the module level instead of the component-level as an optimization then.